lib/fog/vsphere/requests/compute/vm_clone.rb

```diff
@@ -161,8 +161,11 @@ def vm_clone(options = {})
             end
             raise ArgumentError, "domain is required" unless cust_options.key?("domain")
             cust_domain = cust_options['domain']
-            cust_ip_settings = RbVmomi::VIM::CustomizationIPSettings.new(cust_options["ipsettings"]) if cust_options.key?("ipsettings")
-            cust_ip_settings.ip = RbVmomi::VIM::CustomizationFixedIp("ipAddress" => cust_options["ipsettings"]["ip"]) if cust_options.key?("ipsettings")
+            if cust_options.key?("ipsettings")
+              cust_ip_settings = RbVmomi::VIM::CustomizationIPSettings.new(cust_options["ipsettings"])
+              cust_ip_settings.ip = RbVmomi::VIM::CustomizationFixedIp("ipAddress" => cust_options["ipsettings"]["ip"])
+              cust_ip_settings.gateway = cust_options['ipsettings']['gateway'] if cust_options['ipsettings'].key?('gateway')
```

Comment:
What happens if you set `cust_ip_settings.gateway` to `nil`?  I suspect it's a safe operation and that the default is already `nil`.  If that's the case, we can remove the `if` statement modifier and just always assign.